grammar.js

```diff
@@ -215,15 +215,42 @@ module.exports = grammar({
       )
     )),
 
+    _literal: $ => choice(
+      $.symbol,
+      $.integer,
+      $.float,
+      $.boolean,
+      $.nil,
+      $.string
+    ),
+
     symbol: $ => token(seq(':', choice(identifierChars(), operatorChars()))),
+    integer: $ => (/0b[01](_?[01])*|0[oO]?[0-7](_?[0-7])*|(0d)?\d(_?\d)*|0x[0-9a-fA-F](_?[0-9a-fA-F])*/),
+    float: $ => (/\d(_?\d)*\.\d(_?\d)*([eE]\d(_?\d)*)?/),
+    boolean: $ => choice('true', 'false', 'TRUE', 'FALSE'),
+    nil: $ => choice('nil', 'NIL'),
+
+    string: $ => choice(
+      $._single_quoted,
+      $._double_quoted,
+      $._percent_q
+    ),
+    _single_quoted: $ => stringBody("'"),
+    _double_quoted: $ => token(seq('"', repeat(choice(/\\./, $.interpolation, /[^\\"]/)), '"')),
```

Comment:
> Alternatively, could we add a combinator which turns off comment/whitespace parsing within it?

Yeah, I think tree-sitter should definitely have this ultimately. It would fit in to the system well, but isn't a super quick addition.

> It looks like the comment rule gets applied before trying to parse the rules, except within tokens ... Would it be reasonable to instead attempt to parse the comment/whitespace rules when matching the current rule fails?

So currently, both rules (`comment` and `#{`) are tried in parallel, but the comment rule wins because it matches a longer substring of the input. I think that you should be able to assign the `#{` token a higher lexical precedence. Lexical precedence uses the same API as syntactic precedence, but influences the lexer instead of the parser. To apply it, you call the `prec` function *inside* of a call to `token`:

```js
interpolation: $ => seq(
  token(prec(10, '#{')),
  $._expression,
  '}'
)
```

I haven't actually tries this in this grammar though, so I might be missing something.